WooCommerce/Classes/ViewRelated/Products/Edit Product/Subscription/Trial/SubscriptionTrialViewModel.swift

```diff
@@ -0,0 +1,80 @@
+import Foundation
+import Yosemite
+import Combine
+
+/// ViewModel for `SubscriptionTrialView`
+///
+final class SubscriptionTrialViewModel: ObservableObject {
+    @Published var trialLength: String
+
+    @Published var trialPeriod: SubscriptionPeriod
+
+    @Published var isInputValid: Bool = true
+
+    var trialPeriodDescription: String {
+        switch trialLength {
+        case "1":
+            return trialPeriod.descriptionSingular
+        default:
+            return trialPeriod.descriptionPlural
```

Comment:
I noticed this behavior in Core and it's cool that we're replicating this detail 💯